Lib/fontTools/otlLib/builder.py

```diff
@@ -491,6 +493,14 @@ def _makeDeviceTuple(device):
     return (device.DeltaFormat, device.StartSize, device.EndSize,
             tuple(device.DeltaValue))
 
+def _getSinglePosValueSize(valueKey):
+    count = 0
+    for k in valueKey[1:]:
+        if hasattr(k[1], '__len__') and len(k[1]):
+            count += len(k[1][3]) + 3
```

Comment:
The calculation here does not take delta-format into account BTW.

Reply:
oh.. when the device table has DeltaFormat == 0x8000 (aka VariationIndex table), it is always 3-ushort long, right?

Reply:
btw, this `otlLib.buildSinglePos` function is currently only used within feaLib and mtiLib, which only ever output DeltaFormat 1, 2 or 3.

Reply:
The DeltaFormat == 0x8000 is not handled by code in that module that converts Device table to a tuple.  But that's not what I'm talking about.  The code above assumes that each delta value takes 2 bytes.  That's simply not true.  Each delta value takes one of 1, 2, or 4 *bits* depending on DeltaFormat.

Reply:
ok now i'm proper confused, so i'll leave that to you

Reply:
cc @mhosken

Reply:
I don't feel touching it either.  Martin can fix at some point.

Reply:
OK I got it

https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#device-and-variationindex-tables

> Format values 0x0001 to 0x0003 are used for Device tables, and indicate the format of delta adjustment values contained directly within the device table: signed 2-, 4,- or 8-bit values

Martin is just taking the `len()` of the device.DeltaValue (thus assuming one uint16 or 2 bytes per adjustment delta value) but we should instead count the length the packed representation.

As far as I understand, this is not a major issue, we simply don't optimize as much as we could, right?

Reply:
Correct.

Reply:
I agree. I saw this when I wrote it, but decided not to complexify the code to handle a pretty rare use case. If someone who is actually using device records wants to refine the code, they can.